subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheState.kt

```diff
@@ -82,11 +82,14 @@ class ConfigurationCacheState(
     private val codecs: Codecs,
     private val stateFile: ConfigurationCacheStateFile
 ) {
-
-    suspend fun DefaultWriteContext.writeRootBuildState(build: VintageGradleBuild) {
-        writeRootBuild(build)
-        writeInt(0x1ecac8e)
-    }
+    /**
+     * Writes the state for the whole build starting from the given root [build] and returns the set
+     * of stored included build directories.
+     */
+    suspend fun DefaultWriteContext.writeRootBuildState(build: VintageGradleBuild): HashSet<File> =
+        writeRootBuild(build).also {
+            writeInt(0x1ecac8e)
```

Comment:
What is this magic marker?

Reply:
Just a sanity check. We had a bug in the past where the deserialization code was reading less work graph elements than written so I added this marker to ensure defects like that don't go unnoticed.

Reply:
Can you add a comment here so this information is conserved for future readers?

Reply:
Will do!